### ISSUE TITLE
Throw when "Form.props.action" doesn't return a Promise

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/react-advanced-form#157]
## What
Need to throw an error when `Form.props.action()` doesn't return a Promise.

## Why
Right now this will result into `.then() is not a function` error. This may be confusing, and should be handled more precisely by RAF.

## How
```jsx
submit = () => {
  const dispatchedAction = action(...);
  invariant(dispatchedAction instanceof Promise, 'Error message');
}
```

**Comment by kettanaito** [comment]
Started with this task on `dev/form-action-invariant` branch.

**Comment by kettanaito** [comment]
Published under [`0.18.1`](https://github.com/kettanaito/react-advanced-form/commit/47ca1345118f937b7e0167ed5c5e23064a69cdb8). Closing.
